core/drivers/stm32_gpio.c

```diff
@@ -606,6 +725,178 @@ static bool bank_is_registered(const void *fdt, int node)
 	return false;
 }
 
+static TEE_Result apply_rif_config(struct stm32_gpio_bank *bank)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint32_t cidcfgr = 0;
+	unsigned int i = 0;
+
+	if (!bank->rif_cfg)
+		return TEE_SUCCESS;
+
+	if (clk_enable(bank->clock))
+		panic();
+
+	for (i = 0; i < bank->ngpios; i++) {
+		if (!(BIT(i) & bank->rif_cfg->access_mask[0]))
+			continue;
+
+		/*
+		 * When TDCID, OP-TEE should be the one to set the CID filtering
+		 * configuration. Clearing previous configuration prevents
+		 * undesired events during the only legitimate configuration.
+		 */
+		if (bank->is_tdcid)
+			io_clrbits32(bank->base + GPIO_CIDCFGR(i),
+				     GPIO_CIDCFGR_CONF_MASK);
+
+		cidcfgr = io_read32(bank->base + GPIO_CIDCFGR(i));
+
+		/* Check if the controller is in semaphore mode */
+		if (!stm32_rif_semaphore_enabled_and_ok(cidcfgr, RIF_CID1))
+			continue;
+
+		/* If not TDCID, we want to acquire semaphores assigned to us */
+		res = stm32_rif_acquire_semaphore(bank->base + GPIO_SEMCR(i),
+						  GPIO_MAX_CID_SUPPORTED);
+		if (res) {
+			EMSG("Could not acquire semaphore for pin %u", i);
```

Comment:
Suggestion: print also the GPIO bank ID to ease debugging. Eg.
E.g. replace `for pin %u", i)`
with `for pin %C%u", 'A' + bank->bank_id, i)`

This comment applies also to lines 797 and 806.

Reply:
Done with few other occurences as well